db/migrations/manual/20230106-synthetic-runout-file-creation.sql

```diff
@@ -0,0 +1,133 @@
+-- This script will generate synthetic runout file data (cclf_files and cclf_beneficiaries records)
+-- based on the December file of the prior performance year for identified model entity types that
+-- will not have a generated runout file delivered for ingestion by BCDA.
+--
+-- Required: 
+--      (1) Dec 2022 CCLF files have already been delivered & ingested for the model entity type
+--      (2) model entity cannot be termed (is active)
+--
+-- The performance year can be switched in the DECLARE section for reusability, just update the variables.
+
+BEGIN;
+
+DO $$ 
+DECLARE
+model_type VARCHAR(8) := 'Y%'; -- model type to target. KCC model entity type (KCE and KCF) need to do 'C%' and 'K%' respectively
+december_range_start_date cclf_files.created_at%TYPE := '2022-11-30';
+december_range_end_date cclf_files.created_at%TYPE := '2022-12-31';
+file_performance_year cclf_files.performance_year%TYPE := 22;
+synthetic_file_name_convention cclf_files.name%TYPE := CONCAT('.ACO.ZC8R','22','.D','221231','.SYNTHETICRUNOUTFILE');
+
+items RECORD;
+affected_aco_count INTEGER;
+total_synthetic_file_count INTEGER;
+total_mbi_count INTEGER;
+
+BEGIN
```

Comment:
Is this needed when there is a BEGIN on line 11?

Reply:
I typed a lengthy answer for the PR and accidentally deleted the tab instead of clicking back to it :facepalm:

In a nutshell, from what I could find it seems both END and COMMIT for statements that begin with BEGIN do the same thing (but END being a postgres sql extension for COMMIT) in Postgres.
Some years back I worked with SQL Server for a bit and there I believe was a diff between a BEGIN ... END and a BEGIN .. COMMIT. The BEGIN / END was a begin and end to a code block, and BEGIN / COMMIT was used more for beginning and ending transactions.
Esp since doing a DECLARE first with variables used in the script, I then wanted to proceed with a BEGIN ... END to mark the boundaries of the actual execution.